Backpack/Rating/Classes/BPKRating.swift

```diff
@@ -0,0 +1,263 @@
+    }
+
+    private func updateSize() {
+        switch size {
+        case .default:
+            ratingValueLabel.fontStyle = .textLabel1
+            ratingScaleLabel.fontStyle = .textCaption
+            titleLabel.fontStyle = .textLabel1
+            subtitleLabel.fontStyle = .textCaption
+
+            titleSubtitleStackView.axis = .horizontal
+            titleSubtitleStackView.spacing = BPKSpacingMd
+            titleSubtitleStackView.alignment = .lastBaseline
+        case .large:
+            ratingValueLabel.fontStyle = .textHero5
+            ratingScaleLabel.fontStyle = .textBodyDefault
+            titleLabel.fontStyle = .textHeading5
+            subtitleLabel.fontStyle = .textBodyDefault
+
+            titleSubtitleStackView.axis = .vertical
+            titleSubtitleStackView.spacing = .zero
+            titleSubtitleStackView.alignment = .leading
+        }
+    }
+
+    private func displayedValue(from value: Float, scale: BPKRatingScale) -> String {
+        let clampedValue: Float
+        let valueWithinRange = scale.range.contains(value)
+        if !valueWithinRange {
+            if value < scale.range.lowerBound {
+                clampedValue = scale.range.lowerBound
+            } else {
+                clampedValue = scale.range.upperBound
+            }
+        } else {
+            clampedValue = value
+        }
+        return String(format: "%.1f", clampedValue)
+    }
+
+    private func displayedScale(from scale: BPKRatingScale) -> String {
+        return String(format: "/%.0f", scale.range.upperBound)
+    }
+
+    // When size=large and custom subtitleLabel is hidden,
+    // horizontalStackView's alignment = .lastBaseline doesn't work.
+    // To avoid the customTitleView being below the ratingValueLabel's baseline,
+    // an extra constraint is needed to pin the view's bottom to the ratingValueLabel's baseline
+    private func updateCustomTitleViewBottomConstraintIfNeeded() {
+        let isShowingCustomTitleView = titleView != titleLabel
+        let isLargeSize = size == .large
+        let isSubtitleLabelHidden = subtitleLabel.isHidden
+        let isAddedToSuperView = titleView.superview != nil
+
+        if isShowingCustomTitleView && isSubtitleLabelHidden &&
+            isLargeSize && isAddedToSuperView {
+            if titleViewBottomConstraint == nil {
+                titleViewBottomConstraint = titleView.bottomAnchor.constraint(
+                    lessThanOrEqualTo: ratingValueLabel.lastBaselineAnchor
+                )
+                titleViewBottomConstraint?.priority = .required
+            }
+            NSLayoutConstraint.activate([titleViewBottomConstraint!])
+        } else {
+            if
+                let titleViewBottomConstraint = titleViewBottomConstraint,
+                titleViewBottomConstraint.isActive
+            {
+                NSLayoutConstraint.deactivate([titleViewBottomConstraint])
+            }
+        }
```

Comment:
Are we able to always 'set this constraint up' and then activate/deactivate it based on this logic? 
Could be as a lazy var at the top. 

This would simplify the structure here and the code would only execute the first time we call upon it to become active.

Reply:
I will take a look

Reply:
The class now setup CustomTitleView's bottom Constraint during
- init 
- titleView didSet.

The constraint will be activated & deactivated in each `updateLookAndFeel()` call
facf78aae01ab33fdd12c7b7715ee72516b4bca3